core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@KostyaSha In response to your WDTY query, I understand and agree with the sentiment that in general it's not great to have a bunch of trim()s everywhere.  In this case, however, the bug fix is all about trimming and checking the trimming carefully, and I'm afraid we aren't going to get around having to actually trim these strings. :)  I like the suggestions about trying to combine stuff using utility methods, but the !StringUtils.equals(trim(name), trim(newName)) suggested above is actually longer than !newName.trim().equals(name.trim()), so the code as-is seems about as good as we'll be able to get as far as I can see.  If anyone has a better solution that maintains _exactly_ the logic as it is here (which is required to resolve this bug) I'm absolutely willing to update it.